daft/ml/model.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Protocol, runtime_checkable
+
+if TYPE_CHECKING:
+    from daft.ml.typing import Provider
+
+
+@runtime_checkable
+class Model(Protocol):
+    @classmethod
+    def from_provider(cls, provider: Provider, model: str | None = None, **properties: str) -> Model:
+        if provider == "vllm":
+            return cls._from_vllm(model, **properties)
+        elif provider == "sentence_transformers":
+            return cls._from_sentence_transformers(model, **properties)
+        raise ValueError(f"Provider is not yet supported: {provider}")
+
+    @classmethod
```

Comment:
`_from_vllm` and `_from_sentence_transformers` may make more sense as implementations of the Model right?